vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -32,21 +32,27 @@ def _get_inputs_of_controls(page: Page, control_type: ControlType) -> list[State
     """Gets list of `States` for selected `control_type` of triggered `Page`."""
     return [
         State(component_id=control.selector.id, component_property=control.selector._input_property)
-        for control in page.controls
-        if isinstance(control, control_type)
+        for control in cast(Iterable[ControlType], model_manager._get_models(control_type, page))
```

Comment:
Would the `_get_inputs_of_controls` be even more flexible if we searched for all `_filter`/`_parameter` Action functions on the page (instead for `vm.Filter`/`vm.Parameter` components)? In that case if a `_filter` action (or its future public counterpart e.g: `filter`) is assigned to the pure `dcc.Dropdown()`, this component would behave exactly like the `vm.Filter` (except the `_dynamic` behaviour and potentially some other.). 

This is more like a theoretical question and I guess its answer mostly depends on how we see Vizro backend in future and it also depends on many other decisions (e.g. how to make a standalone dcc.Dropdown to be dynamic, and many many more). Currently, I can't say what exactly are all the pros/cons of this approach and is any new use-case covered with this approach or another. So, this comment here is just an idea.

Reply:
Good question! I'm actually thinking about this already in #363 right now and not yet sure or how to handle it, so let me keep this unresolved to come back to.